judo-ui-react/src/main/java/hu/blackbelt/judo/ui/generator/react/UiI18NHelper.java

```diff
@@ -316,4 +322,106 @@ private static Boolean keepTranslationKey(String key) {
         Matcher m = TRANSLATION_KEYS_TO_SKIP.matcher(key);
         return !m.matches();
     }
+
+    public static String magicTranslateFlex(Flex flex) {
+        // currently only used to translate TabController Tabs, because the actual label is inside the first Flex of a Tab
+        PageDefinition page = flex.getPageDefinition();
+        String result = magicTranslatePage(page);
+        result += ".".concat(flex.getName());
+        return transformTranslationKey(result);
+    }
+
+    public static String magicTranslate(VisualElement element) {
+        PageDefinition page = element.getPageDefinition();
+        String result = magicTranslatePage(page);
+
+        if (element instanceof Column) {
+            if (element.eContainer() instanceof ReferenceTypedVisualElement) {
+                ReferenceTypedVisualElement ref = (ReferenceTypedVisualElement) element.eContainer();
+                result += ".".concat(ref.getDataElement().getName());
```

Comment:
this form `result += ".".concat(...);` is a bit confusing for me, but if no one else mentions this, we could leave it as it is.

In our use case `concat(...)` method takes away more than it gives in comparison to `+`, it doesn't give a significant boost to performance.

Reply:
I agree with @bencegelei